### PR TITLE
An RPC connection fix and some vbyte fun

### LINKED ISSUE #3771: Onchain fee roundoff
We seem to be rounding off fees in a way that is incompatible with `bitcoind`: https://github.com/ElementsProject/lightning/issues/3591#issuecomment-641466505 by @whitslack

> I was withdrawing with `feerate=1000perkb`, which should be the minimum-allowed fee rate. Indeed, `bitcoin-cli getmempoolinfo` reports:
>
> ```
> {
>   "loaded": true,
>   "size": 15097,
>   "bytes": 9207924,
>   "usage": 32831760,
>   "maxmempool": 64000000,
>   "mempoolminfee": 0.00001000,
>   "minrelaytxfee": 0.00001000
> }
>```

### PR DESCRIPTION
**Pull request by darosior**
90bce56: I regularly hit this corner case on a bloated-memory node

9230907 and 2fbd490 fix #3771.

Fixes #3771

### CODE REVIEW

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3772/commits/2fbd49061c5b2e3cefea25e9cef2573bd99991ed

File: lightningd/chaintopology.c [hunk 1]
```diff
@@ -474,7 +474,7 @@ u32 feerate_from_style(u32 feerate, enum feerate_style style)
 	case FEERATE_PER_KBYTE:
 		/* Everyone uses satoshi per kbyte, but we use satoshi per ksipa
 		 * (don't round down to zero though)! */
-		return (feerate + 3) / 4;
+		return feerate / 4 + 3;
```
Comment (ZmnSCPxj) [inline lightningd/chaintopology.c:477]: Would this lead to us computing a higher feerate than "Everyone" later on?

Comment (darosior) [reply lightningd/chaintopology.c:477]: I hesitated to use a "return FEERATE_FLOOR if < FEERATE_FLOOR ", but I think we were actually computing *lower* feerate than bitcoind before this.

Comment (ZmnSCPxj) [reply lightningd/chaintopology.c:477]: Okay.  Will ACK this.  Unless you want to change the second commit message.

**Comment by ZmnSCPxj** [comment]
ACK 2fbd49061c5b2e3cefea25e9cef2573bd99991ed

File: lightningd/chaintopology.c (hunk 1 above)
Comment (darosior) [reply lightningd/chaintopology.c:477]: Will need to fixup the tests, so will amend the second commit message while at it.

**Comment by darosior** [comment]
This messed up all feerate-related tests : now I regret not to have made the API in WU for `estimatefees`...

**Comment by darosior** [comment]
Ok, so I went for the "don't go below `FEERATE_FLOOR` approach instead of monkey-patching all the hardcoded feerates in the tests.

**Comment by ZmnSCPxj** [comment]
The linked travis build is passing, but it seems github is not noticing it....

**Pull request merged**